job/tests/test_views.py

```diff
@@ -0,0 +1,266 @@
+        self.assertFalse(job.approved)
+        # Changed fields
+        self.assertEqual(response_data["data"]["cover"], data["cover"])
+        self.assertEqual(response_data["data"]["icon"], data["icon"])
+        self.assertEqual(response_data["data"]["description"], data["description"])
+
+    def test_job_update_view_with_put_method_when_objects_not_exists(self):
+        # Updating non-existent object
+        data = {
+            "id": 10,
+            "name": "برنامه نویس Go",
+            "job_slug": slug_helper("برنامه نویس Go"),
+            "cover": "/default/system/cover2.jpg",
+            "icon": "fas fa-home2",
+            "description": "برنامه نویس Go",
+            "is_deleted": True,
+            "approved": True,
+        }
+        url = "/job/5000/update/"
+        self.client.credentials(HTTP_AUTHORIZATION="JWT " + self.token)
+        response = self.client.put(url, data, format="json")
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(response_data["message"], "Instance does not exist.")
+
+    def test_job_update_view_with_patch_method_when_object_exists(self):
+        # Updating existent object
+        job = Job.objects.first()
+        url = f"/job/{job.id}/update/"
+        partial_data = {
+            "cover": "/default/system/cover3.jpg",
+            "icon": "fas fa-home3",
+            "description": "برنامه نویس Rust",
+        }
+
+        data = {
+            "id": job.id,
+            "name": "برنامه نویس وب",
+            "job_slug": "برنامه-نویس-وب",
+            "cover": "/default/system/cover3.jpg",
+            "icon": "fas fa-home3",
+            "description": "برنامه نویس Rust",
+            "is_deleted": False,
+            "approved": False,
+        }
+        self.client.credentials(HTTP_AUTHORIZATION="JWT " + self.token)
+        response = self.client.patch(url, partial_data, format="json")
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response["content-type"], "application/json")
+        self.assertEquals(response_data["data"], data)
+
+    def test_job_update_with_patch_method_when_object_not_exists(self):
+        # Updating non-existent object
+        partial_data = {
+            "cover": "/default/system/cover3.jpg",
+            "icon": "fas fa-home3",
+            "description": "برنامه نویس Rust",
+        }
+        self.client.credentials(HTTP_AUTHORIZATION="JWT " + self.token)
+        url = "/job/5000/update/"
+        response = self.client.patch(url, partial_data, format="json")
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertEqual(response_data["message"], "Instance does not exist.")
+
+    def test_job_delete_view_when_object_exists(self):
+        # Deleting an existent object
```

Comment:
I think these comments are redundant now, you explained your test functionality on test name

Reply:
exactly.they will be removed